openhands/agenthub/codeact_agent/codeact_agent.py

```diff
@@ -73,24 +75,28 @@ class CodeActAgent(Agent):
 
     def __init__(
         self,
-        llm: LLM,
         config: AgentConfig,
+        llm_config: LLMConfig,
+        llm_registry: LLMRegistry,
```

Comment:
I'd generally expect this to follow a singleton pattern rather than getting passed around